src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -176,6 +176,344 @@ namespace cryptonote
+      { 
+        std::string locked_funds;
+        std::string total_received;
+        std::string total_sent;
+        uint64_t scanned_height;
+        uint64_t scanned_block_height;
+        uint64_t start_height;
+        uint64_t transaction_height;
+        uint64_t blockchain_height;
+        std::list<spent_output> spent_outputs;
+        BEGIN_KV_SERIALIZE_MAP()
+          KV_SERIALIZE(locked_funds)
+          KV_SERIALIZE(total_received)
+          KV_SERIALIZE(total_sent)        
+          KV_SERIALIZE(scanned_height)
+          KV_SERIALIZE(scanned_block_height)
+          KV_SERIALIZE(start_height)
+          KV_SERIALIZE(transaction_height)
+          KV_SERIALIZE(blockchain_height)
+          KV_SERIALIZE(spent_outputs)
+        END_KV_SERIALIZE_MAP()
+      };
+  };
+  
+  //-----------------------------------------------
+  struct COMMAND_RPC_LIGHT_WALLET_GET_UNSPENT_OUTS
+  {
+      struct request
+      {
+        std::string amount;
+        std::string address;
+        std::string view_key;
+
+        BEGIN_KV_SERIALIZE_MAP()
+          KV_SERIALIZE(amount)
+          KV_SERIALIZE(address)
+          KV_SERIALIZE(view_key)
+        END_KV_SERIALIZE_MAP()
+      };
+    
+      
+      struct output {
+        std::string amount;
+        std::string public_key;
+        int index;
+        uint64_t global_index;
+        std::string rct;
+        uint64_t tx_id;
+        std::string tx_hash;
+        std::string tx_pub_key;
+        std::string tx_prefix_hash;
+        std::vector<std::string> spend_key_images;
+        std::string timestamp;
+        uint64_t height;                
+
+
+        BEGIN_KV_SERIALIZE_MAP()
+          KV_SERIALIZE(amount)
+          KV_SERIALIZE(public_key)
+          KV_SERIALIZE(index)
+          KV_SERIALIZE(global_index)
+          KV_SERIALIZE(rct)
+          KV_SERIALIZE(tx_id)
+          KV_SERIALIZE(tx_hash)
+          KV_SERIALIZE(tx_pub_key)
+          KV_SERIALIZE(tx_prefix_hash)
+          KV_SERIALIZE(spend_key_images)  
+          KV_SERIALIZE(timestamp)  
+          KV_SERIALIZE(height)                                    
+        END_KV_SERIALIZE_MAP()
+      };
+      
+      struct response
+      {
+        std::string amount;
+        std::list<output> outputs;
+        BEGIN_KV_SERIALIZE_MAP()
+          KV_SERIALIZE(amount)
+          KV_SERIALIZE(outputs)
+        END_KV_SERIALIZE_MAP()
+      };
+  };
+  
+  //-----------------------------------------------
+  struct COMMAND_RPC_LIGHT_WALLET_GET_RANDOM_OUTS
+  {
+      struct request
+      {
+        std::vector<std::string> amounts;
+        int count;
+
+        BEGIN_KV_SERIALIZE_MAP()
+          KV_SERIALIZE(amounts)
+          KV_SERIALIZE(count)
+        END_KV_SERIALIZE_MAP()
+      };
+    
+      
+      struct output {
+        std::string public_key;
+        std::string global_index; //TODO: why is this a string in this call but not get_unspent_outs.
+        std::string rct;               
```

Comment:
Wouldn't it be better to have three keys, then ? I know I did that kind of thing in the GUI, but that was because Qt makes it very impractical to return more than one result.
As for the question, you can have keys with non canonical names, yes. See my network_address patch, which uses this to provide backward compatibility.